bpf/lib/dbg.h

```diff
@@ -194,6 +195,10 @@ struct debug_capture_msg {
 static __always_inline void cilium_dbg(struct __ctx_buff *ctx, __u8 type,
 				       __u32 arg1, __u32 arg2)
 {
+#ifdef DEBUG_TAGGED
```

Comment:
Instead of introducing yet another #define in the form of `DEBUG_TAGGED`, can we just reuse `CONFIG(tracing_ip_option_type)` to decide whether or not to return here? Better yet, maybe we can just do something like this inside `trace_helpers.h`:

```c
static __always_inline __u64 load_ip_trace_id(void)
{
    return CONFIG(tracing_ip_option_type) ? bpf_trace_id_get() : 0;
}
```

That way, we don't need the embedded `#ifdef` cruft.

In general, we want to use runtime configuration and not introduce new `#define` macros.

Reply:
> That would mean that we can only ever return verbose debug events for tagged packets (unless we split the definitions again like I did in the first iteration of this). So a flow with no ip_trace_id (which is most of them), would immediately return out of this debug function.

Yeah, fair point. Generally, we should aim to use the runtime config `CONFIG(abc)` framework for new config, but in this case avoiding this would require a larger refactoring of the other debug options. So for now let's keep `DEBUG_TAGGED`.